reliability-engineering/terraform/modules/concourse-monitoring/files/dashboards/concourse-slo.json

```
@@ -0,0 +1,417 @@
+            "Sum"
+          ]
+        },
+        {
+          "alias": "",
+          "dimensions": {},
+          "expression": "1 - status5xx / (status2xx+status3xx+status4xx+status5xx)",
+          "id": "errorRatio",
+          "matchExact": true,
+          "metricName": "",
+          "namespace": "",
+          "period": "300",
+          "refId": "C",
+          "region": "default",
+          "statistics": [
+            "Average"
+          ]
+        }
+      ],
+      "timeFrom": null,
+      "timeShift": null,
+      "title": "Availability",
+      "type": "stat"
+    },
+    {
+      "content": "\n# Concourse Availability\n\nWe monitor the ratio of successful vs \nfailure HTTP status\ncodes from the load balancer that sits\nin front of the concourse-web nodes.\n\nThis is a good proxy for if the concourse\nUI/API is up and available for use.\n\n**We aim for 99.9% of requests to\nsucceed over a 7day period.**\n\n\n\n\n",
+      "datasource": null,
+      "fieldConfig": {
+        "defaults": {
+          "custom": {}
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 9,
+        "x": 6,
+        "y": 0
+      },
+      "id": 12,
+      "mode": "markdown",
+      "timeFrom": null,
+      "timeShift": null,
+      "title": "",
+      "type": "text"
+    },
+    {
+      "datasource": null,
+      "description": "Percentage of requests faster than 250ms",
+      "fieldConfig": {
+        "defaults": {
+          "custom": {},
+          "decimals": 2,
+          "mappings": [
+            {
+              "from": "",
+              "id": 0,
+              "operator": "",
+              "text": "",
+              "to": "",
+              "type": 1,
+              "value": ""
+            }
+          ],
+          "noValue": "0",
+          "thresholds": {
+            "mode": "percentage",
+            "steps": [
+              {
+                "color": "dark-red",
+                "value": null
+              },
+              {
+                "color": "dark-yellow",
+                "value": 95
+              },
+              {
+                "color": "dark-green",
+                "value": 99
+              }
+            ]
+          },
+          "unit": "percentunit"
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 6,
+        "x": 0,
+        "y": 8
+      },
+      "id": 2,
+      "options": {
+        "colorMode": "value",
+        "graphMode": "area",
+        "justifyMode": "auto",
+        "orientation": "auto",
+        "reduceOptions": {
+          "calcs": [
+            "mean"
+          ],
+          "fields": "",
+          "values": false
+        }
+      },
+      "pluginVersion": "7.0.5",
+      "targets": [
+        {
+          "expr": "sum(rate(concourse_http_responses_duration_seconds_bucket{le=\"0.25\"}[5m]))\n/\nsum(rate(concourse_http_responses_duration_seconds_count[5m]))\n\n",
```

```suggestion
          "expr": "sum(rate(concourse_http_responses_duration_seconds_bucket{le=\"0.25\",status!="500"}[5m]))\n/\nsum(rate(concourse_http_responses_duration_seconds_count[5m]))\n\n",
```
500s can be fast but they shouldn't count as successful for our SLI.

Future refinement: separate latency targets for `method="GET"` vs `method=~"PUT|POST"`?

---

Sounds like a good idea. I also considered looking into excluding some of the API routes that extremely frequent, but not affected much by performance issues (like serving parts of UI etc, things that barely touch the db).

I like the idea of having some more detailed breakdown graphs next to each of these more broad ones.

---

excluded the 500s